### ISSUE TITLE
In stock status on variable products with stock on product level

### ISSUE THREAD

**Issue by eskadesign** [woocommerce/woocommerce#14880]
### EXPLANATION OF THE ISSUE

Variation products are set to SOLD OUT when activating product level stock management. This type of thing happens when a variation product is set to "Manage stock on product level"
And then in a variation I set these settings: See screenshots

### STEPS TO REPRODUCE THE ISSUE

When I don't set the variations to product level stock, but the parent product has also the 0 in stock and "Allow backorders", then it does says "IN STOCK" on the back end.

### SYSTEM STATUS

<details>
`
### WordPress Environment ###

Home URL: https://www.globalfurniture-elst.nl
Site URL: https://www.globalfurniture-elst.nl
WC Version: 3.0.5
Log Directory Writable: ✔
WP Version: 4.7.4
WP Multisite: –
WP Memory Limit: 4 GB
WP Debug Mode: –
WP Cron: ✔
Language: nl_NL

### Server Environment ###

Server Info: Apache
PHP Version: 7.0.17
PHP Post Max Size: 32 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.36.0
OpenSSL/1.0.1e

SUHOSIN Installed: –
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.5
WC Database Prefix: shpl_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (40) ###

Shield Security: door iControlWP – 5.8.2
Redux Framework: door Team Redux – 3.6.4
Custom Sidebars Pro: door WPMU DEV – 3.0.5
Developer Mode: door Jesper van Engelen – 0.4.1.3
Google Analytics Dashboard voor WP: door Alin Marcu – 4.9.6.2
WPBakery Visual Composer: door Michael M - WPBakery.com – 5.1.1
MainWP Child: door MainWP – 3.3
Mammoth Mega Submenu: door aislin – 1.0.0
Menu Icons: door Dzikri Aziz – 0.10.2
Meta Slider - Pro Addon Pack: door Matcha Labs – 2.6.8
Meta Slider: door Matcha Labs – 3.5.1
Mollie Payments for WooCommerce: door Mollie – 2.5.4
Really Simple SSL: door Rogier Lankhorst – 2.5.13
WPML Multilingual CMS: door OnTheGoSystems – 3.7.0
SW Responsive Slider Widget: door smartaddons.com – 1.0
SW Core: door smartaddons.com – 1.1.0
Sw Woocommerce: door Smartaddons – 1.1.0
TinyMCE Advanced: door Andrew Ozz – 4.5.6
User Role Editor: door Vladimir Garagulya – 4.32.3
WooCommerce Availability Notifications: door ThemePlugger – 1.1.5
WooCommerce Multilingual: door OnTheGoSystems – 4.1.3
WooCommerce PDF Invoices: door Bas Elbers – 2.8.1
WooCommerce Sequential Order Numbers: door SkyVerge – 1.8.1
WooCommerce Tab Manager: door SkyVerge – 1.8.0
WooCommerce Table Rate Shipping: door Bolder Elements – 3.6.6
WooCommerce: door Automattic – 3.0.5
WooSwipe: door Dean Oakley
Eric Jinks
BJ CJ – 1.1.6.5

Yoast SEO Premium: door Team Yoast – 4.7
WP Add Custom CSS: door Daniele De Santis – 1.0.1
WP Fastest Cache Premium: door Emre Vona – 1.3.4
WP Fastest Cache: door Emre Vona – 0.8.6.8
WP Hummingbird: door WPMU DEV – 1.4.3
WPML CMS Nav: door OnTheGoSystems – 1.4.20
WPML Media: door OnTheGoSystems – 2.2.0
WPML Sticky Links: door OnTheGoSystems – 1.4.1
WPML String Translation: door OnTheGoSystems – 2.5.3
WPML Translation Management: door OnTheGoSystems – 2.3.0
WPML Widgets: door Jeroen Sormani – 1.0.6
WPMU DEV Dashboard: door WPMU DEV – 4.3
Yoast SEO: WooCommerce: door Team Yoast – 4.7

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: EUR (€)
Currency Position: left_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
simple (simple-fr)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Winkelbasis: #10 - /shop/
Winkelmand: #11 - /winkelmand/
Afrekenen: #12 - /afrekenen/
Mijn account: #13 - /mijn-account/

### Theme ###

Name: Furnicom Child Theme
Version: 1.5.0
Author URL: http://flytheme.net/
Child Theme: ✔
Parent Theme Name: Furnicom
Parent Theme Version: 1.6.0
Parent Theme Author URL: http://flytheme.net/
WooCommerce Support: ✔

### Templates ###

Overrides: furnicom-child-theme/woocommerce/archive-product.php
furnicom/woocommerce/cart/cart.php
furnicom/woocommerce/content-product.php
furnicom/woocommerce/content-single-product.php
furnicom/woocommerce/loop/loop-end.php
furnicom/woocommerce/loop/loop-start.php
furnicom/woocommerce/loop/pagination.php
furnicom/woocommerce/loop/price.php
furnicom/woocommerce/loop/rating.php
furnicom/woocommerce/myaccount/form-login.php
furnicom-child-theme/woocommerce/single-product/add-to-cart/grouped.php
furnicom/woocommerce/single-product/add-to-cart/simple.php
furnicom/woocommerce/single-product/meta.php
furnicom/woocommerce/single-product/price.php
furnicom/woocommerce/single-product/product-image.php
furnicom/woocommerce/single-product/product-thumbnails.php
furnicom/woocommerce/single-product/rating.php
furnicom/woocommerce/single-product/share.php
furnicom/woocommerce/single-product/tabs/tabs.php
furnicom/woocommerce/single-product/title.php
furnicom/woocommerce/single-product-reviews.php
furnicom/woocommerce/single-product.php

`
</details>
![global-var](https://cloud.githubusercontent.com/assets/17767488/25704012/340576c2-30d8-11e7-9eb3-7e8011830659.jpg)
![global-var-stock](https://cloud.githubusercontent.com/assets/17767488/25704011/33fe447e-30d8-11e7-80cc-8f6453a1fca1.jpg)

**Comment by melindahelt** [comment]
Confirmed on my test site as well as 569721-zd-woothemes

**Comment by Pab89** [comment]
I am having the same issue

**Comment by mikejolley** [comment]
If stock management is enabled at parent level, it will **not** consider all of the child stock levels if those are enabled also. If you're managing stock levels per variation as per the screenshots, the parent level should not be set nor enabled, then it will respect the levels and stock statuses of the variations themselves.

Parent stock management should only be enabled if you have several variations which share the same stock qty.

If parent stock management is on, the stock status and display of the stock level will always be that of the parent.

Hope that makes sense. I know it's a bit confusing and something the team are looking to simplify in future updates.

**Comment by TomasHurtz** [comment]
This is causing me a major headache - because of the way it was set up previously, we had all our Parent products set as follows; 

Manage stock?  YES Enable stock management at product level
Stock Quantity: 0

Then in each variable item, we managed the child stock inventory as per correct stock levels.  We then communicate all of this to Xero accounting system via a cloud app.  Xero then shows the parent product, followed by the variations underneath.

Part of the reason for the "Parent" to be set at 0 was because the entire range of 1000 products was imported using WPAllImport - which recommended this method to list the products.

This change seems to be rather spurious and I didn't see any documentation / notification it would change.  So now we need to redesign our entire stock system to suit a Woocommerce update??
This will take days of work!!

Is there a way to revert to the previous system?

**Comment by WPprodigy** [comment]
@mikejolley This is also affecting some extensions where inventory is synced externally. eBay Sync, Shipwire, etc. These extensions need "manage stock" enabled at the product level even if some variations have their own stock as well.

Just thought I'd mention it.

**Comment by TomasHurtz** [comment]
Yes @WPprodigy - this is the issue I have - Xero accounting system needs a parent product - at least in our case. 

Also - another thing I've noticed is - In WP Admin > Products - any parent product that has an out of stock variation now shows "Sold Out" - which is very misleading.  It also showed the same on front end, until I disable the Sold Out sticker.

**Comment by mikejolley** [comment]
Sorry but this is a bug fix and should not be reverted. The option is clearly labeled: 

> Enable stock management at product level

If you're managing stock of variations, you're not managing this at product level, so it's wrong. **Managing stock at product level prohibits us from syncing the stock status with the children.**

> I don't know all the details of "why", but has to do with the ability to push inventory levels into the product level if it needs to

This is also wrong and can cause bugs. e.g. 2 variations with backorders enabled. 1 has 10 in stock. The other is backordered and has -5 in stock. Is stock 5? No.

Plugins relying on this bug should either; a) sync the variations, not the parent, or b) handle variable products differently to loop over children to get the total stock levels needed rather than looking at 'manage stock'.

I'll leave this open for @bekarice to comment, but I really do not want to revert this and will point out, the parent stock management option is confusing and fully expect this to be phased out at some point in the future.

**Comment by bekarice** [comment]
@mikejolley as far as I can see, this change is almost purely presentational in the admin, no?

1. Setup a variable product, each variation should manage stock + have some stock. 
1. On the inventory tab for the parent, set "Manage stock" and leave stock qty 0 for the parent

The product is viewable and purchasable on the frontend because the variations have stock. This is true in WC 2.5 through 3.x, regardless of backorder status.

The only issues I see are:

1. If "Don't allow backorders" is set on the parent, it shows as "out of stock" in the admin product list. However, the product is still purchasable in the shop. In WooCommerce 2.6 and before, it just showed "In Stock" (no product count).
1. If you hide out of stock items from the shop, despite this being purchasable, the product will be hidden as I'm assuming it checks the parent stock status. In WC 2.6 and before, it would still be shown (likely tied to the above point).

So functionally, almost nothing has changed between these versions; having stock management enabled on the parent with stock = 0 doesn't prevent any purchasing on the frontend in either version, and only changes frontend behavior in WC 3.0+ with backorders disabled + hide out of stock items. 

Not sure what the bug fix was intended to do -- override variation stock if parent stock was managed? Block purchasing if so? Mostly, in the way it stands now, it's just confusing for the merchant to see "Out of stock" in the admin when really it's not out of stock, but purchasing is still possible since all the children are purchasable.

Now with plugins like Shipwire, the "Shipwire managed?" option is set on the parent product for variable products as a whole for a few reasons:

1. Adding this into the variations UI puts yet another option into an already crammed and difficult-to-understand space for new merchants
1. There is almost no case we've come across in which a merchant wants a specific variation managed by a fulfillment company rather than the product as a whole, so it's not needed on a per-variation basis.
1. Enabling this on the parent product, even if parent stock was 0, had no downsides before (as products were always listed in stock + purchasable), but it also showed people it was a required configuration if they didn't read the docs.

With that said, we can look to un-force stock management on the parent and potentially add a notice on product save instead with recommended settings since now the admin display of this stock status is different / the product could be hidden from the catalog, which didn't happen before.
 
However, on the WC core side, I think this UI as it stands would benefit from an improvement in the next version since it's super confusing right now -- do I manage inventory on the product? variation? should I enter both -- my parent should be the sum of the variations? 

I'd also be hesitant to re-do a bunch of variation handling in inventory management extensions only to have to re-do it again when the variations UI changes, especially given it would be done to work around what amounts to not much more than a presentational bug in the admin right now.

Since you can do one **OR** the other -- manage variations as a whole, or manage variations individually -- why not make this a choice instead of checkboxes on both the parent and the variations?

**Option 1**: Manage variations individually -- show "Manage stock" checkbox in the variations UI

![screen shot 2017-05-19 at 4 00 14 pm](https://cloud.githubusercontent.com/assets/4983547/26264913/936e60e4-3cac-11e7-8581-6a0b38eabf97.png)

**Option 2**: Manage variations as a group (on the parent) -- do not show "Manage stock" options in the variations UI

![screen shot 2017-05-19 at 4 01 24 pm](https://cloud.githubusercontent.com/assets/4983547/26264923/a3330912-3cac-11e7-9ce8-489b93e98661.png)

This could be a stepping stone to make the inventory management choice clearer right now before re-doing variations handling completely.

**Comment by TomasHurtz** [comment]
I have to agree with @bekarice - the UI is confusing.  Good input, thank you.

There has not really been any good instruction on how inventory options should be used - so I feel making this "carte blanche" change is rather unfair, inappropriate and without notice.  If I had known that this change was coming, I could set up my 000's of products to use manage stock at the parent level in the appropriate way.

As it stands, i'm in limbo and waiting on a decision as a result of this discussion.  Right now, I have all my products listed in XERO under the parents which is not now working - so no stock management.  Giving the user the option of whether to turn this on or off would be preferable to me, whether by way of a function hook / filter or other method.

**Comment by mikejolley** [comment]
@bekarice lets add your proposal to dev chat agenda.

**Comment by bekarice** [comment]
👍  have a partner call overlapping with dev chat this week but will try to keep a pulse on it or add thoughts afterwards

**Comment by mikejolley** [comment]
Hmm @bekarice Just noticed:

> Since you can do one OR the other -- manage variations as a whole, or manage variations individually -- why not make this a choice instead of checkboxes on both the parent and the variations?

Thats not the case either right now... you can use both, and any un-variation stock managed products will inherit. But the parent does not inherit stock levels from children.

**Comment by bekarice** [comment]
@mikejolley I'd wonder how common this is:

> Thats not the case either right now... you can use both, and any un-variation stock managed products will inherit. But the parent does not inherit stock levels from children.

I didn't even know you could do that 😜  but I agree that it could likely be removed. It seems the parent stock is best for a variation that "inherits" from a parent, such as blank tee shirt with an iron-on design (different variations) -- the stock is still the blank tee shirt for all designs. However, having a mix of stock management -- some designs with their own stock, some inheriting from a blank tee -- seems like it would be non-standard, and they probably would be better served as separate products.

You'll likely have a small set of people using it the way you've outlined, but I think a change to either / or for how stock is managed would be beneficial for more merchants / easier to understand overall

**Comment by stircrazi** [comment]
I am not a developer, just a guy trying to run a website with Woocommerce for eBay. Right after this latest update I noticed that any new additions or even revisions to existing inventory. On the Woocommerce side (products) it would show the item was out of stock. I knew this to be incorrect. After a little research we discovered that by unchecking the "managed stock" this problem went a way. With that said, a new problem arose. Inventories are no longer syncing between Woocommerce and eBay. This is a huge problem for us, with over 300 variable listings and close to 10,000 variations that  we are now having to manually inventory again. But, since there isn't a resolution yet. Can't fix inventory until issue with Managed Stock is resolved.

**Comment by thomasmery** [comment]
at the moment it is, imho, quite confusing when a Variable Product is set to let its Variations manage stock  - UI-wise at least 

it is my understanding that there is a _sync_ mechanism that will set the parent product stock status based on the variations stock status i.e. :
- Parent Product will be set as 'outofstock' when all variations are 'outofstock'

In addition to @bekarice proposal (see screenshots above) would it be possible to at least make this clearer on the Parent Product Inventory tab by adding messages, something along the lines of: 

- variations managed
> This variable product stock status is currently managed through its variations stock statuses. This Product Stock Status is currently: _out of stock_

- globally managed
> This variable product stock status is set globally for all variations

it would be great to see this clarification (at least some sort of clarification) before more in-depth changes

also, if someone can point to hooks that allow adding such messages on specific installs it would help in the meantime

thanks!

**Comment by eduardoalfaro** [comment]
I have the same issue, so confused but "by-passed" by now... ;-D also, please, keep an eye on stock column at woocommerce products page. If you are working with "parent" stock, the stock number will show, but if you are wokring with only "variations" stock, you will not have the number of units on stock, intead, you will have only the stock available message... It will be have so convenient to know the total stock for all variations... Any suggestions or thoughts about this issue? Thank you so much!!!

**Comment by LSB-Web-Design** [comment]
Any suggestions on how to untick the Enable stock management at product level option for all products? This is causing major issues for a number of sites I manage so we will need to turn this option off for all products. This could take days if we do it manually.

**Comment by WPprodigy** [comment]
@LSB-Web-Design there is quick edit/bulk edit that can do this: http://cld.wthms.co/vFVR4

The new product import/exporter that will be in the next WC core release ~tomorrow could also help.

**Comment by mikejolley** [comment]
I've folded this into https://github.com/woocommerce/woocommerce/issues/13085 The option solution above could work, however, it does stop a current use case so it technically breaking. We should look at this from scratch as we intended.
